logdnaHandlerTest.py

```diff
@@ -0,0 +1,152 @@
+import concurrent.futures
+import json
+import threading
+import time
+
+from http.server import BaseHTTPRequestHandler,HTTPServer
+import logging
+import unittest
+
+from logdna import LogDNAHandler
+
+current_milli_time = lambda: int(round(time.time() * 1000))
+
+key = '< YOUR INGESTION KEY HERE >'
+log = logging.getLogger('logdna')
+log.setLevel(logging.INFO)
+
+options = {
+  'hostname': 'localhost',
+  'url': 'http://localhost:8081',
+  'ip': '10.0.1.1',
+  'mac': 'C0:FF:EE:C0:FF:EE'
+}
+
+expectedLines = []
+class successful_RequestHandler(BaseHTTPRequestHandler):
+  def do_POST(self):
+        content_length = int(self.headers['Content-Length'])
+        body = self.rfile.read(content_length)
+        self.send_response(200)
+
+        self.end_headers()
+        body = json.loads(body)['ls']
+        for keys in body:
+            expectedLines.append(keys['line'])
+
+class failed_RequestHandler(BaseHTTPRequestHandler):
+  def do_POST(self):
+        content_length = int(self.headers['Content-Length'])
+        body = self.rfile.read(content_length)
+        self.send_response(400)
+        self.end_headers()
+
+class LogDNAHandlerTest(unittest.TestCase):
+    def server_recieves_messages(self):
+        options = {
+          'hostname': 'localhost',
+          'url': 'http://localhost:8081',
+          'ip': '10.0.1.1',
+          'mac': 'C0:FF:EE:C0:FF:EE'
+        }
+
+        successRes = True
+        server_address = ('localhost', 8081)
+        httpd = HTTPServer(server_address, successful_RequestHandler)
+
+        test = LogDNAHandler(key, options)
+        log.addHandler(test)
+        line = "python python python"
+
+        def send_log():
+            log.info(line)
+
+        serverThread = threading.Thread(target=httpd.handle_request)
+        logdnaThread = threading.Thread(target=send_log)
+        serverThread.daemon = True
+        logdnaThread.daemon = True
+
+        serverThread.start()
+        logdnaThread.start()
+
+        serverThread.join()
+        logdnaThread.join()
+
+        self.assertTrue(len(expectedLines) == 1)
+        self.assertTrue(len(test.buf) == 0)
+        self.assertTrue(line in expectedLines)
```

Comment:
Just a note that when Python is 3 only, this can be `assertIn`. Since we're not there yet, we can keep this as is.